testreporters/ocr.go

```diff
@@ -0,0 +1,268 @@
+		log.Error().Err(err).Msg("Error sending slack message to webhook")
+	}
+
+	log.Info().Msg("OCR Soak Test Report")
+	log.Info().Msg("--------------------")
+	for contractAddress, report := range o.Reports {
+		log.Info().
+			Str("Contract Address", report.ContractAddress).
+			Uint("Total Rounds Processed", report.TotalRounds).
+			Str("Average Round Time", fmt.Sprint(report.averageRoundTime)).
+			Str("Longest Round Time", fmt.Sprint(report.LongestRoundTime)).
+			Str("Shortest Round Time", fmt.Sprint(report.ShortestRoundTime)).
+			Uint("Average Round Blocks", report.averageRoundBlocks).
+			Uint("Longest Round Blocks", report.LongestRoundBlocks).
+			Uint("Shortest Round Blocks", report.ShortestRoundBlocks).
+			Msg(contractAddress)
+	}
+	log.Info().Msg("--------------------")
+}
+
+// UpdateReport updates the report based on the latest info
+func (o *OCRSoakTestReport) UpdateReport(roundTime time.Duration, blockLength uint) {
+	// Updates min values from default 0
+	if o.ShortestRoundBlocks == 0 {
+		o.ShortestRoundBlocks = blockLength
+	}
+	if o.ShortestRoundTime == 0 {
+		o.ShortestRoundTime = roundTime
+	}
+	o.TotalRounds++
+	o.totalRoundTimes += roundTime
+	o.totalBlockLengths += blockLength
+	if roundTime >= o.LongestRoundTime {
+		o.LongestRoundTime = roundTime
+	}
+	if roundTime <= o.ShortestRoundTime {
+		o.ShortestRoundTime = roundTime
+	}
+	if blockLength >= o.LongestRoundBlocks {
+		o.LongestRoundBlocks = blockLength
+	}
+	if blockLength <= o.ShortestRoundBlocks {
+		o.ShortestRoundBlocks = blockLength
+	}
+}
+
+// writes a CSV report on the test runner
+func (o *OCRSoakTestReporter) writeCSV(csvPath string) error {
+	ocrReportFile, err := os.Create(filepath.Join(utils.ProjectRoot, "ocr_soak_report.csv"))
+	if err != nil {
+		return err
+	}
+	defer ocrReportFile.Close()
+
+	ocrReportWriter := csv.NewWriter(ocrReportFile)
+	err = ocrReportWriter.Write([]string{
+		"Contract Index",
+		"Contract Address",
+		"Total Rounds Processed",
+		"Average Round Time",
+		"Longest Round Time",
+		"Shortest Round Time",
+		"Average Round Blocks",
+		"Longest Round Blocks",
+		"Shortest Round Blocks",
+	})
+	if err != nil {
+		return err
+	}
+	for contractIndex, report := range o.Reports {
+		err = ocrReportWriter.Write([]string{
+			fmt.Sprint(contractIndex),
+			report.ContractAddress,
+			fmt.Sprint(report.TotalRounds),
+			fmt.Sprint(report.averageRoundTime),
+			fmt.Sprint(report.LongestRoundTime),
+			fmt.Sprint(report.ShortestRoundTime),
+			fmt.Sprint(report.averageRoundBlocks),
+			fmt.Sprint(report.LongestRoundBlocks),
+			fmt.Sprint(report.ShortestRoundBlocks),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	ocrReportWriter.Flush()
+
+	log.Info().Str("Location", filepath.Join(utils.ProjectRoot, "ocr_soak_report.csv")).Msg("Wrote CSV file")
+	return nil
+}
+
+// sends a slack message to a slack webhook if one is provided
+func (o *OCRSoakTestReporter) sendSlack(csvLocation, namespace string) error {
```

Comment:
Do we want this function to be moved to a more common part of the library since there is nothing in here specific to ocr and it would make sending to slack easier for other test setups?

Reply:
Ya, was recently thinking about that since it would be helpful in the keeper tests we want to run.We can tack it on to this PR, or the next one that will move the performance tests to also use the remote runner.